src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.cs

```diff
@@ -182,7 +182,7 @@ private void UnwrapTaskIfNeeded(SemanticModel model, ref ITypeSymbol returnType)
                 var taskType = model.Compilation.TaskType();
                 if (originalDefinition.Equals(taskType))
                 {
-                    returnType = model.Compilation.GetSpecialType(SpecialType.System_Void);
```

Comment:
To be honest, I'm OK with is actually trying to be resilient to this rather than the OrThrow if you don't want to expose the method, but this is fine too.